roofit/hs3/src/RooJSONFactoryWSTool.cxx

```diff
@@ -1584,6 +1589,60 @@ bool RooJSONFactoryWSTool::importYML(std::string const &filename)
    return this->importYML(infile);
 }
 
+void RooJSONFactoryWSTool::importFunctionfromString(const std::string &jsonString){
+   std::stringstream is(jsonString);
+   std::unique_ptr<JSONTree> tree = JSONTree::create(is);
+   const JSONNode &n = tree->rootnode();
+
+   // Check if function is a p.d.f. 
+   bool isDistribution = endsWith(n["type"].val(), "_dist");
+
+   if(isDistribution){
+      this->importFunction(n, true, false);
+   } else {
+      this->importFunction(n, false, false);
+   } 
```

Comment:
```suggestion
   std::stringstream is(jsonString);
   this->importFunction(JSONTree::create(is)->rootnode(), false);
```

There is something I noticed now, thanks to your work here. The `importFunction()` right now takes an argument `isPdf`, but actually it could just figure out if a given function is ` pdf by this `_distr` suffix, just like we are now doing in your new function!

So what about changing the signature of `importFunction()` to:
```c++
void RooJSONFactoryWSTool::importFunction(const JSONNode &p, bool importAllDependants)
```
And then, in that funciton, you'll do:
```c++
   // Check if function is a p.d.f.
   bool isPdf = endsWith(p["type"].val(), "_dist");+   // Check if function is a p.d.f.
```
Like this, the caller of `importFunction()` never needs to pass this `isPdf` argument, and your new `importFunctionfromString()` would simplify to what I suggested above.

Reply:
That's a nice idea! I've added this in the modified commit